tests/api_workflow/test_api_workflow_compute_worker.py

```diff
@@ -669,3 +667,45 @@ def test__snake_to_camel_case() -> None:
     assert _snake_to_camel_case("lorem_ipsum") == "loremIpsum"
     assert _snake_to_camel_case("lorem_ipsum_dolor") == "loremIpsumDolor"
     assert _snake_to_camel_case("loremIpsum") == "loremIpsum"  # do nothing
+
+
+def test__validate_config(mocker: MockerFixture) -> None:
+
+    mock_obj = mocker.MagicMock()
+    mock_obj.swagger_types = {
+        "config": "str",
+        "nested_config": mocker.MagicMock(),
+    }
+
+    _validate_config(
+        cfg={
+            "config": "hello world",
+            "nested_config": {
+                "config": "xyz"
+            }
+        },
+        obj=mock_obj
+    )
+
+
+def test__validate_config__typo(mocker: MockerFixture) -> None:
+
+    mock_obj = mocker.MagicMock()
+    mock_obj.swagger_types = {
+        "config": "str",
+        "nested_config": mocker.MagicMock(),
+    }
+    # We need to explicitly delete it from the mocked object otherwise hasattr
+    # behaves strangely.
+    del mock_obj.configx
+
+    with pytest.raises(
+        InvalidConfigurationError,
+        match="Option `configx` does not exist!"
+    ):
+        _validate_config(
+            cfg={
+                "configx": "hello world",
+            },
+            obj=mock_obj
+        )
```

Comment:
We should add more tests:
* A test on a real swagger class
* A test for a misspelling of a nested field

Reply:
I replaced the mocked tests with real swagger classes.